game-of-life-native/Game of Life/GameOfLifeViewController.swift

```diff
@@ -11,59 +11,104 @@ import Foundation
+        let loop = SwiftGameLoop() as KGOLGameLoop
+        let model = KGOLBoardModelImpl(initialBoard: boardEntity, gameLoop: loop)
+        boardPresenter = KGOLBoardPresenter(boardModel: model)
+
         super.init(coder: aDecoder)
+        controlButton.addTarget(self, action: #selector(controlButtonAction), for: .touchUpInside)
+        patternSelectionButton.addTarget(self, action: #selector(patternSelectionButtonAction), for: .touchUpInside)
     }
-    
+
+    private func setupLayout() {
+        let yOffset: CGFloat = CGFloat(35)
+
+        controlButton.frame = CGRect(
+                x: 0,
+                y: yOffset,
+                width: self.view.bounds.width / 2,
+                height: yOffset)
+
+        patternSelectionButton.frame = CGRect(
+                x: self.view.bounds.width / 2,
+                y: yOffset,
+                width: self.view.bounds.width / 2,
+                height: yOffset)
+        patternSelectionButton.setTitle("Choose a Pattern", for: .normal)
+
+        boardView.backgroundColor = .black
+        boardView.frame = CGRect(
+                x: 0,
+                y: controlButton.bounds.height + yOffset,
+                width: self.view.bounds.width,
+                height: self.view.bounds.width)
+    }
+
     func renderControlButtonLabel(controlButtonLabel: String) {
-        controllButton.setTitle(controlButtonLabel, for: .normal)
+        controlButton.setTitle(controlButtonLabel, for: .normal)
     }
-    
+
     func renderPatternSelectionVisibility(visibility: Bool) {
-        
+        patternSelectionButton.isHidden = !visibility
     }
-    
+
     func renderBoardWith(boardViewInput: KGOLBoardViewInput) {
         boardView.renderBoard(boardViewInput: boardViewInput)
     }
-    
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        if (appPresenter == nil){
-            let model = KGOLAppModel()
-            appPresenter = KGOLAppPresenter(model: model)
-            controllButton.backgroundColor = .green
-            controllButton.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
-            self.view.addSubview(controllButton)
-            
-            boardView.backgroundColor = .blue
-            self.view.addSubview(boardView)
-        }
-        
-        appPresenter?.bind(view: self)
-        boardView.willAppear()
+        view.addSubview(controlButton)
+        view.addSubview(patternSelectionButton)
+        view.addSubview(boardView)
     }
-    
-    @objc func buttonAction(sender: UIButton!) {
-        onControlButtonClicked()
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        setupLayout()
+        appPresenter.bind(view: self)
+        boardPresenter.bind(boardView: boardView)
     }
-    
+
+    @objc func controlButtonAction(sender: UIButton!) {
+        let _ = onControlButtonClicked()
```

Comment:
Kotlin native compiles kotlin closures to a closure that returns `KGOLStdlibUnit`. To suppress the warning of the unused result we to assign it to `_`

Reply:
could Koitlin return a @discardable result?

Reply:
Kotlin doesn't know about anything `discardable` since it's a swift feature.

Reply:
that was my question, then your solution is the correct one! 👍